added Support for Authorized Key Options

eni23 opened the pull request:
again: added options "options" and "ssh_from" in authorized_keys, so keys get recognized if options or from is changed on host's authorized_keys manually. now with language cleanup :)

b6d committed:
postgresql_db: add args lc_collate, lc_ctype

Allow the specification of additional locale settings (lc_collate and lc_ctype) when creating a new database (state=present).

Fail if the specified database already exists with different locale/encoding settings. (These settings can't be changed for existing databases as far as I know, and failing seems better than suggesting that no change was necessary by returning changed=False)

dagwieers committed:
Make private key customizable per host using ansible_ssh_private_key_file configurable as variable

This fixes issue #1026 for @toshywoshy together with ansible_ssh_user and ansible_ssh_port.

eni23 committed:
added Support for Authorized Key Options

eni23 committed:
added Support for Authorized Key Options

gregswift committed:
Extend authorized_keys module for use with alternate AuthorizedKeysFile configurations

bcoca committed:
now ssh connection plugin will only pass user as an option to ssh if it is
differnt from the current user. This should enable overrides for user in
.ssh/config w/o breaking any current functionality.

Signed-off-by: Brian Coca <briancoca+dev@gmail.com>

junegunn committed:
terminate immediately on a failure

junegunn committed:
compare the number of available hosts before and after running each task

Commit:
Initial commit of rax library

This library provides functionality for the Rackspace Public Cloud by
way of the official pyrax SDK (https://github.com/rackspace/pyrax). At
this time only the cloudservers service is functional. Instances can be
created or deleted. Idempotency is provided on matching instances with
the same name, flavor, image, and metadata values within a given
region.

pyrax usage does require a credentials file written out to hold username
and API key.  See pyrax documentation for details
(https://github.com/rackspace/pyrax/blob/master/docs/pyrax_doc.md)

Commit:
Initial commit of rax inventory plugin

The rax inventory plugin provides a way to discovery inventory in the
Rackspace Public Cloud by way of pyrax, the official SDK. Grouping will
be done if a group:name is found in the instance metadata. When a single
host is queried all the instance details are returned with a rax_
prefix.

Because inventory plugins cannot take extra arguments, ENV variables
must be set to point to the pyrax compatible credentials file and the
region to query against.

Commit:
Fix up docs and add ability to insert files

Files can be inserted during server creation (like a fully formed
authorized_keys file). This code allows that to happen.

Docs were updated for formatting, location, and to add the new entry for
files.

Commit:
Adding npm module

joelsemar committed:
adds private_ip param to ec2 module

lwade committed:
Updated and improved the documentation (examples)

magnusart committed:
Added functionality for md5 checksum together with a new overwrite param
to indicate that remote file should be overwritten if a file with the
same filename exists.

bcoca committed:
added missing build deps for debian/ubuntu to readme
Signed-off-by: Brian Coca <briancoca+dev@gmail.com>

jochu committed:
mysql_db: use --password= instead of -p in dump/import

mavimo committed:
moved dependencies in the right place

mavimo committed:
Moved choises in the right place

gw0 committed:
Improved backrefs example in lineinfile.

willthames committed:
Added DEFAULT_EXECUTABLE as a constant

Use DEFAULT_EXECUTABLE when no executable is passed to
_low_level_command_exec

Works as a standard constant - can be overridden in all the normal ways
and defaults to /bin/sh

Motiviation is for a user that only has /bin/bash in /etc/sudoers

mpdehaan committed:
Fix typo in message.

mpdehaan committed:
Merge pull request #2563 from gw0/fix-lineinfile

Improved backrefs example in lineinfile.

skvidal committed:
simple fix for issue 2564
when we encounter a group we've not seen before - we need
to make sure we make it a subgroup of 'all', no matter what

chrishoffman committed:
Updating version added, tweaking instantiation

Tinche committed:
Changed the service module to terminate early if only changing the enabled state.
Expanded the documentation slightly.

Tinche committed:
Additional example in service documentation.

willthames committed:
Documented executable setting for ansible.cfg

gw0 committed:
Improved Lookup plugins documentation

Commit:
Merge branch 'any-errors-fatal' of git://github.com/junegunn/ansible into devel

Commit:
Merge branch 'ssh_config_user_overrides' of git://github.com/bcoca/ansible into devel

Commit:
Disable a test temporarily

Commit:
Standardize test output

Commit:
Update changelog

mpdehaan committed:
Merge pull request #2537 from j2sol/rax

Provide functionality to work with Rackspace's public cloud

Commit:
Changelog update

Commit:
Work in progress.

mpdehaan committed:
Merge pull request #2542 from chrishoffman/npm

Adding npm module

Commit:
update changelog

mpdehaan committed:
Merge pull request #2571 from Tinche/service

Service module changed to terminate early if only enabled specified

mpdehaan committed:
Merge pull request #2552 from lwade/ec2docs

Updated and improved the documentation (examples)

mpdehaan committed:
Merge pull request #2557 from bcoca/deb_make_reqs

small doc fix for 'make deb'

mpdehaan committed:
Merge pull request #2558 from jochu/mysql-password

mysql_db: use --password= instead of -p in dump/import

mpdehaan committed:
Merge pull request #2559 from mavimo/feature/documentation-vagrant

vagrant module documentation

eni23 committed:
minor language cleanup as wished

mpdehaan committed:
Merge pull request #2565 from willthames/default_exec

Added DEFAULT_EXECUTABLE as a constant

Commit:
organize changelog

mpdehaan committed:
Merge pull request #2570 from skvidal/devel

simple fix for issue 2564

mpdehaan committed:
Merge pull request #2446 from b6d/postgresql_db-module-extension

postgresql_db: add args lc_collate, lc_ctype

mpdehaan commented:
The new options also need to have a version_added attribute in the code so we can tell when those two params were added

version_added: "1.2"

Can you please update this?  Thanks!

eni23 committed:
updated documentation

mpdehaan committed:
Merge pull request #2575 from gw0/docs-lookup

Improved Lookup plugins documentation

Commit:
Standardize args/kwargs on connection plugins.

Commit:
Chroot also needs args/kwargs.

Commit:
update changelog

Commit:
Merge branch 'devel' of git://github.com/gregswift/ansible into devel

Commit:
Fixup module docs.

mpdehaan committed:
Merge pull request #2578 from arrfab/selinux-fix-2

Added an extra check for target nodes with selinux but without libselinux-python package installed

Commit:
This SELinux commit is too strict and we should only fail if trying to read/tweak SELinux attributes else
python-selinux can't be installed w/ the yum module when missing.

Revert "Added an extra check for target nodes with selinux but without libselinux-python package installed"

This reverts commit 550986b6a1a0af30294c16bb3155150de8851d7d.

mpdehaan commented:
Thanks, unfortunately not your lucky day yet :)  Sorry about this.

There is another change I just merged that requires a rebase on this.

Can you rebase your changes on top of the recent addition?  The other patch came in before yours.

Thanks!

Then we'll be done, I promise!  :)

mpdehaan committed:
Merge pull request #2550 from joelsemar/devel

adds private_ip param to ec2 module

mpdehaan committed:
Merge pull request #2554 from magnusart/s3-overwrite-param

New try: s3 overwrite and checksum support

Commit:
Add version added info for s3 module.

Commit:
Fixup ec2 doc formatting and use complex args format to break up long lines.

Commit:
Fixup authorized key doc formatting

eni23 committed:
Merge branch 'devel' of github.com:adfinis-sygroup/ansible into devel

Conflicts:
	library/authorized_key

mpdehaan commented:
whoa, this is full of merge commits now, which I can't take.

You need to do "git pull --rebase" vs "git pull"

or "git rebase" vs "git merge".

The pull request should basically contain _just_ your changes.

mpdehaan commented:
if you want, you should be able to set up a new branch and then cherry pick your commits over to the new one and then _rebase_ (not merge) against mine.
